src/lib.rs

```diff
@@ -136,6 +144,98 @@ where
     }
 }
 
+impl<P, T> Atom<P>
+where
+    P: IntoRawPtr + FromRawPtr + Deref<Target = T>,
+{
+    /// Stores a value into the pointer if the current value is the same as the
```

Comment:
Stores `new` in this atom if `current` has the same raw pointer representation as the currently stored value.